backend/db/analysis.py

```diff
@@ -56,20 +56,19 @@ def _current_counts(
             df = pd.DataFrame(counts)
             df = df[df['schema'] == schema]
             return df
-    # Get tables
-    with get_db_connection(schema=schema, local=local) as con:
-        tables: List[str] = list_tables(con, filter_temp_refresh_tables=filter_temp_refresh_tables)
+    # Get previous counts
     with get_db_connection(schema='', local=local) as con:
-        # Get previous counts
         timestamps: List[datetime] = [
-            dp.parse(x[0]) for x in sql_query(con, f'SELECT DISTINCT timestamp from counts;', return_with_keys=False)]
+            dp.parse(x[0]) for x in sql_query(con, f"SELECT DISTINCT timestamp from counts WHERE schema = '{schema}';", return_with_keys=False)]
         most_recent_timestamp: str = str(max(timestamps)) if timestamps else None
         prev_counts: List[Dict] = [dict(x) for x in sql_query(con, f'SELECT * from counts;', return_with_keys=True)]
         prev_counts_df = pd.DataFrame(prev_counts)
-        # Get counts
+    # Get current counts / deltas
+    with get_db_connection(schema=schema, local=local) as con:
+        tables: List[str] = list_tables(con, filter_temp_refresh_tables=filter_temp_refresh_tables)
         table_rows: Dict[str, Dict[str, Any]] = {}
         for table in tables:
-            count: int = [x for x in sql_query(con, f'SELECT COUNT(*) from n3c.{table};', return_with_keys=False)][0][0]
+            count: int = [x for x in sql_query(con, f'SELECT COUNT(*) from {schema}.{table};', return_with_keys=False)][0][0]
```

Comment:
## The essential bugfix
We / I have suspected for a few months now that this is broken. I'm surprised it went unnoticed, but after a while I started to suspect something was wrong because the deltas were always 0, and sometimes I wasn't running the it immediately after the backup procedure.

<details><summary>Verification</summary>
<p>

I verified that the results are expected. I added 1 row to `junk`, and it detected.

[df.csv](https://github.com/jhu-bids/TermHub/files/14324687/df.csv)
<img width="337" alt="Screenshot 2024-02-18 at 5 26 49 PM" src="https://github.com/jhu-bids/TermHub/assets/13045020/dc869eae-99ff-4133-b1ce-6b04d88043ed">


</p>
</details>